src/protocols/abstract/Base.ts

```diff
@@ -1,20 +1,50 @@
 'use strict';
 
-const KuzzleError = require('../../KuzzleError');
-const { uuidv4 } = require('../../utils/uuidv4');
-const { KuzzleEventEmitter } = require('../../core/KuzzleEventEmitter');
-const PendingRequest = require('./PendingRequest');
-
-class KuzzleAbstractProtocol extends KuzzleEventEmitter {
-  constructor (host, options = {}, name = undefined) {
+import { KuzzleError } from '../../KuzzleError';
+import { uuidv4 } from '../../utils/uuidv4';
+import { KuzzleEventEmitter } from '../../core/KuzzleEventEmitter';
+import { PendingRequest } from './PendingRequest';
+import { KuzzleRequest, JSONObject } from '../../utils/interfaces';
+
+/**
+ * @todo proper TS conversion
+ */
+export abstract class KuzzleAbstractProtocol extends KuzzleEventEmitter {
+  private _pendingRequests: Map<string, PendingRequest>;
+  private _host: string;
+  private _name: string;
+  private _port: number;
+  private _ssl: boolean;
+
+  public id: string;
+
+  public state: string;
+
+  constructor (host: string, options: JSONObject = {}, name: string = undefined) {
     super();
 
     this._pendingRequests = new Map();
     this._host = host;
     this._name = name;
     const port = parseInt(options.port, 10);
     this._port = isNaN(port) ? 7512 : port;
-    this._ssl = typeof options.sslConnection === 'boolean' ? options.sslConnection : false;
+
+    if (options.ssl !== undefined && options.sslConnection !== undefined) {
+      throw new Error('Both "ssl" and "sslConnection" options are set. Use only "ssl".');
```

Comment:
Why don't we just trigger a `console.warn` when `sslConnection` is set, saying it's going to be deprecated in the next version?

Reply:
IMHO we should do this for every deprecated feature used by users but also we don't want 1) flood the console 2) print messages in production so even if in this case the message won't be printed twice (only at instantiation), this is a bigger subject about how we warn users when they use deprecated features.

@rolljee work on the deprecation module on Kuzzle will be the first step, then we will have to think about how to integrate deprecation notifications in the SDK for developers

Reply:
Plus, this error is thrown if BOTH "ssl" and "sslConnection" are set, because in that case, and if they have different value, then the SDK cannot decide what to do. So the best course of action is to prevent both those options to be set at the same time.

Reply:
I remember VueJS showing deprecation warning in development builds and hiding them in production builds. But we don't have access to such a notion, do we?

Reply:
Maybe we could since we use the same tools as any other frontend build
In Node environment we could rely on NODE_ENV